modules/public/logic.py

```diff
@@ -281,3 +281,44 @@ def external(attrs, new=False):
         formatted_paragraphs.append(paragraph.replace('\n', '<br />'))
 
     return formatted_paragraphs
+
+
+def convert_channel_maps(channel_maps_list):
+    """
+    Converts channel maps list to format easier to manipulate
+
+    Example:
+    - Input:
+    [
+      {
+        'architecture': 'arch'
+        'map': [{'info': 'release', ...}, ...],
+        'track': 'track 1'
+      },
+      ...
+    ]
+    - Output:
+    {
+      'arch': {
+        'track 1': [{'info': 'release', ...}, ...],
+        ...
+      },
+      ...
+    }
+
+    :param channel_maps_list: The channel maps list returned by the API
+
+    :returns: The channel maps reshaped
+    """
+    channel_maps = {}
+    for channel_map in channel_maps_list:
+        arch = channel_map['architecture']
+        track = channel_map['track']
+        channel_maps[arch] = {}
```

Comment:
Does this work properly?
If there are 2 tracks for same architecture the second one will remove the first one if I read the code correctly.

Reply:
I QA'ed it and it doesn't work for example for data like this:

```
    channel_map = [
        {
            'architecture': 'arch',
            'map': [
                {'info': 'release', 'channel': 'stable'},
                {'info': 'release', 'channel': 'beta'}
            ],
            'track': 'latest'
        },
        {
            'architecture': 'arch',
            'map': [
              {'info': 'release', 'channel': 'stable'},
              {'info': 'release', 'channel': 'beta'}
            ],
            'track': 'track2'
        },
        ]
```

Reply:
done